### DISCUSSION TITLE
4 relays / 4 switches.  HomeKit turns on relays but switches do nothing.  Script checked/double checked HELP!

### DISCUSSION THREAD

**Discussion by RKF-FKR** [RavenSystem/esp-homekit-devices#2114] [Q&A] [answered]
Cannot get these buttons to do anything.  ESP32.  Pullups on all button inputs - set to mode 6. 
Can somebody please look at this script and guide me?

4 relays 
4 switches/buttons

{"c":{"o":6,"io":[[[2,13,14,12,27],2],[[26,25,33,32],6,1],[[0],6]],"l":2,"b":[[0,5]]},"a":[{"t":2,"0":{"r":[[13,0]]},"1":{"r":[[13,1]],"b":[[26]]},"es":[{"t":2,"0":{"r":[[14,0]]},"1":{"r":[[14,1]],"b":[[25]]}},{"t":2,"0":{"r":[[12,0]]},"1":{"r":[[12,1]],"b":[[33]]}},{"t":2,"0":{"r":[[27,0]]},"1":{"r":[[27,1]],"b":[[32]]}}]}]}

I can get it to work without the "es" but I want to learn what I am doing wrong.  Working version below.

{"c":{"io":[[[12,13,14,27],2],[[25,26,32,33],6,1],[[0],6]],"o":6},"a":[{"t":2,"0":{"r":[[12,0]]},"1":{"r":[[12,1]]},"b":[[25]],"s":0},{"t":2,"0":{"r":[[13,0]]},"1":{"r":[[13,1]]},"b":[[26]],"s":0},{"t":2,"0":{"r":[[14,0]]},"1":{"r":[[14,1]]},"b":[[32]],"s":0},{"t":2,"0":{"r":[[27,0]]},"1":{"r":[[27,1]]},"b":[[33]],"s":0}]}

**Comment by Tellicious** [comment, marked as answer]
Are you sure that the GPIO0 button doesn't need a pull-up and to be inverted in the first script? Otherwise it will enter setup mode every time you reset the device

Try `[[0],6,1,1]` instead of `[[0],6]`

**Reply by RKF-FKR** [reply]
noted, cheers.

**Reply by Tellicious** [reply]
Can you please mark this as answered, If the question is solved?
